lib/node_modules/@stdlib/assert/is-same-array-like-object/test/test.js

```diff
@@ -0,0 +1,107 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2024 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var tape = require( 'tape' );
+var isSameArrayLikeObject = require( '@stdlib/assert/is-same-array-like-object/lib' );
+
+
+// TESTS //
+
+tape( 'main export is a function', function test( t ) {
+	t.ok( true, __filename );
+	t.strictEqual( typeof isSameArrayLikeObject, 'function', 'main export is a function' );
+	t.end();
+});
+
+tape( 'the function returns `true` if provided two array-like values having the same contents', function test( t ) {
+	var x;
+	var y;
+
+	x = [ 1.0, 2.0, 3.0 ];
+	t.strictEqual( isSameArrayLikeObject( x, x ), true, 'returns expected value' );
+
+	x = [ 1.0, 2.0, 3.0 ];
+	y = [ 1.0, 2.0, 3.0 ];
+	t.strictEqual( isSameArrayLikeObject( x, y ), true, 'returns expected value' );
+
+	x = [ -0.0, 0.0, -0.0 ];
+	y = [ -0.0, 0.0, -0.0 ];
+	t.strictEqual( isSameArrayLikeObject( x, y ), true, 'returns expected value' );
+
+	x = [ NaN, NaN, NaN ];
+	y = [ NaN, NaN, NaN ];
+	t.strictEqual( isSameArrayLikeObject( x, y ), true, 'returns expected value' );
+
```

Comment:
Let's add a test case for an array-like object that is not a generic array:

```suggestion
	x = [ NaN, NaN, NaN ];
	y = [ NaN, NaN, NaN ];
	t.strictEqual( isSameArrayLikeObject( x, y ), true, 'returns expected value' );

	x = { 'length': 10 };
	y = { 'length': 10 };
	t.strictEqual( isSameArrayLikeObject( x, y ), true, 'returns expected value' );
```